### DISCUSSION TITLE
Configured JDK runtime not used for implicit maven profile activation

### DISCUSSION THREAD

**Discussion by caymaynard** [redhat-developer/vscode-java#3818] [General]
I'm not entirely sure whether something is missing in my configuration, or whether there is an issue in how the plugin calls maven (or m2e)
I have my project configured for Java 11, with the settings like 
```
 "java.configuration.runtimes": [
    {
      "path": "/opt/jdk-11",
      "name": "JavaSE-11"
    }
  ],
```
This works for most of my projects, but I have one which has a dependency on `com.helger:ph-jaxb-pom:pom:1.0.0`. That particular version does not work on anything past JDK 11. It has a maven profile with the activation `<jdk>[9,12)</jdk>` (https://github.com/phax/ph-jaxb-pom/blob/ph-jaxb-pom-1.0.0/pom.xml#L93C9-L93C26)

I would expect this to work with my configuration, but instead, it fails to load this dependency, because the maven profile is not activated.
Is this failing to activate because my runtime is not used, or does the language server use its own configured runtime for all the maven tasks?

**Comment by snjeza** [comment]
m2e/maven uses the runtime jdk (>=17) to check the jdk property. 
You can try to use `com.helger:ph-jaxb-pom:pom:1.0.3`

**Reply by caymaynard** [reply]
Thank you @snjeza!
That was the cause of the issue, forcing `com.helger:ph-jaxb-pom:pom:1.0.3` to be used got the extension working.
